src/core/qgsabstractdatabaseproviderconnection.h

```diff
@@ -0,0 +1,406 @@
+
+    /**
+     * Checks whether a table \a name exists in the given \a schema
+     * \throws QgsProviderConnectionException
+     */
+    virtual bool tableExists( const QString &schema, const QString &name ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Drops a vector (or aspatial) table with given \a schema (schema is ignored if not supported by the backend) and \a name
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void dropVectorTable( const QString &schema, const QString &name ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Drops a raster table with given \a schema (schema is ignored  if not supported by the backend) and \a name
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void dropRasterTable( const QString &schema, const QString &name ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Renames a table with given \a schema (schema is ignored  if not supported by the backend) and \a name
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void renameTable( const QString &schema, const QString &name, const QString &newName ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Creates a new schema \a schema
+     * \throws QgsProviderConnectionException
+     */
+    virtual void createSchema( const QString &name ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Drops an entire \a schema
+     * \param force if TRUE, a DROP CASCADE will drop all related objects
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void dropSchema( const QString &name, bool force = false ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Renames a \a schema
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void renameSchema( const QString &name, const QString &newName ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Executes raw \a sql and returns the (possibly empty) list of results in a multi-dimensional array
+     * \throws QgsProviderConnectionException
+     */
+    virtual QList<QList<QVariant>> executeSql( const QString &sql ) const SIP_THROW( QgsProviderConnectionException );
```

Comment:
QList< QVariantList >

Reply:
Yes, I started with QVariantList and QVariantMap but I've got SIP errors, I'll try that again.

Reply:
Nope, sip is NOT happy.